pkg/virt-controller/watch/vmi.go

```diff
@@ -2131,3 +2149,77 @@ func (c *VMIController) handleDynamicInterfaceRequests(vmi *virtv1.VirtualMachin
 
 	return nil
 }
+
+func (c *VMIController) updateInterfaceStatus(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) error {
```

Comment:
I think most of the content here needs to move to a different file or to the network package.

Due to prev disagreement about this specific topic, I can do this myself as a follow up.